homeassistant/components/overkiz/__init__.py

```diff
@@ -144,3 +197,25 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data[DOMAIN].pop(entry.entry_id)
 
     return unload_ok
+
+
+async def async_remove_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Remove local token if applicable, before removing the Config Entry."""
+
+    if token_uuid := entry.data.get(CONF_TOKEN_UUID):
+        # Create Overkiz cloud client for removal of the local token
+        username = entry.data[CONF_USERNAME]
+        password = entry.data[CONF_PASSWORD]
+        server = SUPPORTED_SERVERS[entry.data[CONF_SERVER]]
+
+        session = async_create_clientsession(hass)
+        client = OverkizClient(
+            username=username, password=password, session=session, server=server
+        )
```

Comment:
Maybe extract this in a separate function since you're doing this twice in this file?